ddtrace/tracer/span.go

```diff
@@ -424,7 +424,7 @@ func (s *span) Finish(opts ...ddtrace.FinishOption) {
 		if !cfg.FinishTime.IsZero() {
 			t = cfg.FinishTime.UnixNano()
 		}
-		if cfg.Error != nil {
+		if cfg.Error != nil && !reflect.ValueOf(cfg.Error).IsNil() {
```

Comment:
Summary of offline discussion:

Let's move this to `WithError` so we avoid adding a malformed/invalid error value to `cfg.Error` to begin with. Otherwise the pose the risk of someone else using this `cfg.Error` value somewhere else in the code that's unsafe without realizing they need to do this weird reflect non-nil check.